File: z_cs_py_yapilanlar/CSharedMemory.cs
public class CSharedMemory
{
    string SonOkumaZamani;
    string SonYazmaZamani;
    string SonResetZamani;
    string SonClearZamani;

    string SonOkumaZamaniKey;
    string SonYazmaZamaniKey;
    string SonResetZamaniKey;
    string SonClearZamaniKey;
    /*
    var Anahtar11 = Sistem.Name + ";" + Sistem.Sembol + ";" + Sistem.Periyot + ";" + "YAZMA ZAMANI";
    var Anahtar12 = Sistem.Name + ";" + Sistem.Sembol + ";" + Sistem.Periyot + ";" + "RESET ZAMANI";
    var Anahtar13 = Sistem.Name + ";" + Sistem.Sembol + ";" + Sistem.Periyot + ";" + "OKUMA ZAMANI";
    var Anahtar12 = Sistem.Name + ";" + Sistem.Sembol + ";" + Sistem.Periyot + ";" + "CLEAR ZAMANI";
    */
    ~CSharedMemory()
    {

    }

    public CSharedMemory()
    {

    }

    /*public CSharedMemory Initialize(dynamic Sistem)
    {
        return this;
    }

    public CSharedMemory Reset(dynamic Sistem)
    {
        return this;
    }*/

    public string CreateKey(dynamic Sistem, string Suffix)
    {
        return ( Sistem.Name + ";" + Sistem.Sembol + ";" + Sistem.Periyot + ";" + Suffix );
    }

    public bool IsMemoryNull(dynamic Sistem, string Key)
    {
        return (Sistem.NesneGetir(Key) == null);
    }

    public void ClearMemory(dynamic Sistem, string Key)
    {
        SonClearZamaniKey = Key + ";" + "CLEAR ZAMANI";
        SonClearZamani = DateTime.Now.ToString("HH:mm:ss"); //yyyy.MM.dd HH:mm:ss
        Sistem.NesneKaydet(SonClearZamaniKey, SonClearZamani);

        Sistem.NesneKaydet(Key, null);
    }

    public void ResetMemory(dynamic Sistem, string Key)
    {
        SonResetZamaniKey = Key + ";" + "RESET ZAMANI";
        SonResetZamani = DateTime.Now.ToString("HH:mm:ss"); //yyyy.MM.dd HH:mm:ss
        Sistem.NesneKaydet(SonResetZamaniKey, SonResetZamani);

        Sistem.NesneKaydet(Key, null);
    }

    public void WriteToMemory(dynamic Sistem, string Key, dynamic Value)
    {
        SonYazmaZamaniKey = Key + ";" + "YAZMA ZAMANI";
        SonYazmaZamani = DateTime.Now.ToString("HH:mm:ss"); //yyyy.MM.dd HH:mm:ss
        Sistem.NesneKaydet(SonYazmaZamaniKey, SonYazmaZamani);

        Sistem.NesneKaydet(Key, Value);
    }

    public dynamic ReadFromMemory(dynamic Sistem, string Key)
    {
        SonOkumaZamaniKey = Key + ";" + "OKUMA ZAMANI";
        SonOkumaZamani = DateTime.Now.ToString("HH:mm:ss"); //yyyy.MM.dd HH:mm:ss
        Sistem.NesneKaydet(SonOkumaZamaniKey, SonOkumaZamani);

        return Sistem.NesneGetir(Key);
    }

    public dynamic ReadFromMemoryAsString(dynamic Sistem, string Key)
    {
        return ReadFromMemory(Sistem, Key);
    }

    public dynamic ReadFromMemoryAsInteger(dynamic Sistem, string Key)
    {
        return Convert.ToInt32( ReadFromMemory(Sistem, Key) );
    }

    public dynamic ReadFromMemoryAsSingle(dynamic Sistem, string Key)
    {
        return Convert.ToSingle( ReadFromMemory(Sistem, Key) );
    }

    public dynamic ReadFromMemoryAsDouble(dynamic Sistem, string Key)
    {
        return Convert.ToDouble( ReadFromMemory(Sistem, Key) );
    }

    /*
    var LastExecutionId = 0;
    var LastExecutionKey = Sistem.Name + ";" + Sistem.Sembol + ";" + Sistem.Periyot + ";" + "LastExecutionId";

    var LastExecutionIdStr = Sistem.NesneGetir(LastExecutionKey);
    if (LastExecutionIdStr == null) {
        LastExecutionId = 0;
        Sistem.NesneKaydet(LastExecutionKey, LastExecutionId);
    }
    else {
        LastExecutionId = Convert.ToInt32(LastExecutionIdStr);
        LastExecutionId++;
        Sistem.NesneKaydet(LastExecutionKey, LastExecutionId);
    }
    */
}

File: z_cs_py_yapilanlar/CSharedMemory.py
import datetime

class CSharedMemory:
    def __init__(self):
        self.SonOkumaZamani = None
        self.SonYazmaZamani = None
        self.SonResetZamani = None
        self.SonClearZamani = None
        self.SonOkumaZamaniKey = None
        self.SonYazmaZamaniKey = None
        self.SonResetZamaniKey = None
        self.SonClearZamaniKey = None

    def __del__(self):
        pass

    def create_key(self, Sistem, Suffix):
        return f'{Sistem.Name};{Sistem.Sembol};{Sistem.Periyot};{Suffix}'

    def is_memory_null(self, Sistem, Key):
        return Sistem.NesneGetir(Key) is None

    def clear_memory(self, Sistem, Key):
        self.SonClearZamaniKey = f'{Key};CLEAR ZAMANI'
        self.SonClearZamani = datetime.datetime.now().strftime("%H:%M:%S")
        Sistem.NesneKaydet(self.SonClearZamaniKey, self.SonClearZamani)
        Sistem.NesneKaydet(Key, None)

    def reset_memory(self, Sistem, Key):
        self.SonResetZamaniKey = f'{Key};RESET ZAMANI'
        self.SonResetZamani = datetime.datetime.now().strftime("%H:%M:%S")
        Sistem.NesneKaydet(self.SonResetZamaniKey, self.SonResetZamani)
        Sistem.NesneKaydet(Key, None)

    def write_to_memory(self, Sistem, Key, Value):
        self.SonYazmaZamaniKey = f'{Key};YAZMA ZAMANI'
        self.SonYazmaZamani = datetime.datetime.now().strftime("%H:%M:%S")
        Sistem.NesneKaydet(self.SonYazmaZamaniKey, self.SonYazmaZamani)
        Sistem.NesneKaydet(Key, Value)

    def read_from_memory(self, Sistem, Key):
        self.SonOkumaZamaniKey = f'{Key};OKUMA ZAMANI'
        self.SonOkumaZamani = datetime.datetime.now().strftime("%H:%M:%S")
        Sistem.NesneKaydet(self.SonOkumaZamaniKey, self.SonOkumaZamani)
        return Sistem.NesneGetir(Key)

    def read_from_memory_as_string(self, Sistem, Key):
        return str(self.read_from_memory(Sistem, Key))

    def read_from_memory_as_integer(self, Sistem, Key):
        return int(self.read_from_memory(Sistem, Key))

    def read_from_memory_as_single(self, Sistem, Key):
        return float(self.read_from_memory(Sistem, Key))

    def read_from_memory_as_double(self, Sistem, Key):
        return float(self.read_from_memory(Sistem, Key))
